### PR TITLE
nmc/add new unit test files

### PR DESCRIPTION
**Pull request by AnkurBavisa**
Added new Unit Test files(which are not present in Nextcloud repo) with template, dependent header files and some member access dependency.
Later we will add testcase for NMC changes & nextcloud can also add required test cases.

### CODE REVIEW

**Comment by mgallien** [comment]
@AnkurBavisa could you please rebase this branch on top of the master branch that is the target of the PR ?
there are currently duplicated commits that are also present in the target branch ?

File: test/testusermodel.cpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+#include "localdiscoverytracker.h"
+#include "syncrunfilelog.h"
+#include "syncengine.h"
+#include <QtTest>
+
+#include "account.h"
+#define private public
+#include "accountstate.h"
+#include "folder.h"
+#include "gui/tray/UserModel.h"
```
Comment (mgallien) [inline test/testusermodel.cpp:17]: this should be in lower case

File: test/testwindow.cpp [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ */
+
+#include <QQmlApplicationEngine>
+#include <qqmlengine.h>
+#include <QQmlContext>
+#include <QQuickItem>
+#include <QtTest>
+
+#include "tray/UserModel.h"
```
Comment (mgallien) [inline test/testwindow.cpp:14]: this should be in lower case

File: test/testusermodel.cpp [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ *
+ */
+
+#include "localdiscoverytracker.h"
+#include "syncrunfilelog.h"
+#include "syncengine.h"
+#include <QtTest>
+
+#include "account.h"
+#define private public
```
Comment (mgallien) [inline test/testusermodel.cpp:14]: if I remember correctly, on windows `public` or `private` symbols in a shared library are encoded with different symbols
is this really needed and working ?

**Review by mgallien — CHANGES_REQUESTED** [review]
we were unable to compile the branch due to errors with include of headers

File: test/testusermodel.cpp (hunk 3 above)
Comment (AnkurBavisa) [reply test/testusermodel.cpp:14]: Okay, sure.

**Comment by AnkurBavisa** [comment]
> @AnkurBavisa could you please rebase this branch on top of the master branch that is the target of the PR ? there are currently duplicated commits that are also present in the target branch ?

Okay, sure

File: test/testwindow.cpp (hunk 2 above)
Comment (AnkurBavisa) [reply test/testwindow.cpp:14]: Okay,

File: test/testusermodel.cpp (hunk 1 above)
Comment (AnkurBavisa) [reply test/testusermodel.cpp:17]: Okay

**Comment by AnkurBavisa** [comment]
> Use rebase to update it to master commits, not merge: git checkout master git pull origin master --rebase git checkout add_new_unit_test_files git rebase master

Yes, we tried same, but its still showing, locally showing only 2 commits while rebasing but in PR again its showing 25+ commits, so decided to create new PR.

**Comment by AnkurBavisa** [comment]
Closing this PR due to too many unresolved commits, raise new PR for the same.
New PR: https://github.com/nextcloud/desktop/pull/5867

**Pull request closed without merging**